libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
There is potential starvation of the _next_ set of on-demand requests that come in, since some workers will be tied up with the previous prefetch requests already.  Also, extra background CPU and network usage.  I don't think it's a big deal, just want to make sure we acknowledge it somewhere and are aware of it.

This is still way better than the current situation.